src/GameNode.js

```diff
@@ -8,8 +8,12 @@ const gameNode = (color, onClick, coordinates2d, border, fill, text, asset, play
 };
 
 class Shape {
-    constructor(color, shapeType, shapeInfo, playerIds, onClick, effects, input) {
-        this.node = gameNode(color, onClick, shapeInfo.coordinates2d, shapeInfo.border, shapeInfo.fill, null, null, playerIds, effects, input);
+    constructor(args) {
```

Comment:
I feel like we should destructure `arg` here into something like 
```
{ coordinates2d , shapeType, color, onClick, border, fill, playerIds, effects, input, ...rest }
```

I think this would allow us to take in additional properties someone wanted to add to the gamenode, so after we construct the node and add the `.id` property
we could then add on any additional properties that were passed into via that new rest argument


I dunno if what I wrote makes sense, but lemme know